gem/lib/metasploit-payloads/crypto.rb

```diff
@@ -0,0 +1,44 @@
+require 'openssl'
+
+module MetasploitPayloads
+  module Crypto
+    CIPHER_NAME = 'chacha20'.freeze
+    IV = 'EncryptedPayload'.freeze # 16 bytes
+    KEY = 'Rapid7MetasploitEncryptedPayload'.freeze # 32 bytes
+    ENCRYPTED_PAYLOAD_HEADER = 'encrypted_payload_chacha20_v1'.freeze
```

Comment:
Since this is a string value it should probably be updated to always occupy a set length or, alternatively have a format that is intended to be parsable. Padding this to a set header length like 64 bytes would be sufficient and ensure there's not a collision in the future with the name of `encrypted_payload_chacha20_v10`.